src/content/blog/val-town-newsletter-15.md

```diff
@@ -0,0 +1,230 @@
+
+https://status.val.town/
+
+![Screenshot of the new Val Town status page](./val-town-newsletter-15/status-page.png)
+
+https://x.com/Andre_Terron/status/1750578181600649312?s=20
+
+### TypeScript-driven autocompletion
+
+https://github.com/val-town/val.town/pull/3645
+
+### CodeMirror Continue for comments
+
+https://x.com/tmcw/status/1745147192656957771?s=20
+
+### Improved Feedback button
+
+https://x.com/stevekrouse/status/1749800319217700990?s=20
+
+### Add code folding
+
+- TODO ensure we closed this github discussion and link to it here
+
+### Misc updates
+
+- Fixed the big, old, annoying flicker on page load
+- Redesigned our scheduled val & cron UI
+- "Run Now" button on scheduled and cron vals
+- Tabs on the bottom of vals for viewing logs and previewing HTTP vals
+- Fix flicker bug with our AI autocomplete
+- New comments on a val notify anyone who commented on that val
+- HTTP responses can return up to 10Mb now (40x more than before)
+- Val readmes can be edited via API
+- Add URL ?search data to HTTP logs
+- Updated Deno to 1.40.2 (there's now [an API for getting this version data](https://www.val.town/v/std/deno_version)!)
+- Fix [val.town/brand](https://val.town/brand) assets
+- Defer loading dprint to speed up page loading times
+- Fix scheduled val bugs
+- Include val readmes in val pull requests
+- Add `VALTOWN_API_URL` environment variable to enable std library usage locally
+- Deprecate showing output in favor of console logs
+- Forked vals default to the name of the original val
+- Renamed "Secrets" to "Environment Variables"
+- Improved error message to when users try to preview a private HTTP val
+- Speed up querying for the latest evaluation of a val (4 seconds to 40ms)
+- [Halved our database size](https://x.com/tmcw/status/1742559580255658462?s=20) by removing stale data
+- iframe previews of other peoples' HTTP vals are closed by default
+- Disable grammarly correction in the editor
+- [Improved Google indexing by adding pagination and noindex on vals without readmes](https://x.com/tmcw/status/1753458573764337783?s=20) (thanks [@tomcritchlow](https://www.val.town/u/tomcritchlow))
+- Limit the number of versions a val can have to 14,400 (10 days in minutes)
+- Fix bugs with our homepage buttons
+
+### Roadmap
+
+#### Feburary
+
+- AI autocomplete improvements
+- Improved val run scalability and pricing
+- Improved workflows for POST-request HTTP vals
+- Improved /trending ranking
+- Button to request someone to publish a val
+- Job posting & careers page
+- New intro video
+- Better onboarding for new users
+- Drop "untitled\_" from new val names
+
+### Soon-ish
+
+- Speed up val execution
+- Improved val organization in sidebar (TODO link to feature request)
+- Val Town Projects (for more complex projects, branching and pull requests as groups of vals)
+- TODO pull more from feature requests
+
+## Kind words
+
+- https://x.com/meekaale/status/1740818276152996271?s=20
```

Comment:
I'm not sure how to handle this section. One idea is to screenshot all of these kind words and make a little wall of them. Basically like we have on our logged out homepage:

<img width="854" alt="Screenshot 2024-02-09 at 13 41 32@2x" src="https://github.com/val-town/val-town-blog/assets/2288939/8ce69868-36d8-49e9-bb33-091bef6eeef5">

Reply:
I really like that idea. Do we have a component for that or should we port it here?

Reply:
```ts
function TestimonialCard({
  name,
  avatar,
  quote,
  reference,
}: {
  name: string;
  avatar: string;
  quote: string;
  reference?: string;
}) {
  return (
    <div className="bg-white text-gray-700 border-amber-300 relative rounded-2xl border-2 p-4">
      <blockquote className="pb-2" cite="url">
        {quote}
      </blockquote>
      <div className="flex items-center space-x-4 pt-2">
        <img
          src={avatar}
          alt="avatar"
          loading="lazy"
          className="border-gray-200 inline h-10 w-10 shrink-0 rounded-full border"
        />
        <figcaption>
          <div className="text-gray-700">{name}</div>
          <cite className="text-gray-400 text-sm italic">{reference}</cite>
        </figcaption>
      </div>
    </div>
  );
```

Reply:
Ok. The biggest challenge here (in my opinion) is the container that lays those cards out (the CSS). If you could share that part I can make an Astro component out of those and try to put that into the post.

Reply:
```ts
<div className="bg-amber-50 border-amber-300 w-full border-b-2 border-t-2 py-20">
        <div className="mx-auto w-full max-w-6xl px-6">
          <div className="w-full">
            <div className="flex items-center justify-items-center pb-10">
              <HeartIcon className="text-rose-400 m-2 h-8 w-8" />
              <h2 className="text-gray-900 text-2xl font-semibold sm:text-4xl">
                Kind words
              </h2>
            </div>
            <div className="grid grid-cols-1 gap-x-6 sm:grid-cols-3">
              <div className="space-y-8">
                <TestimonialCard
                  name="Wes Bos"
                  quote={`It's been a while since I've seen something as neat at val.town\n
            It's serverless functions meets myspace? You have to try it.\n
            Just start coding - nothing gets in your way. variables persist, great for iOT, cron jobs, npm support...`}
                  avatar={wesbos}
                  reference="Syntax.fm host, on Twitter"
                  // url="https://twitter.com/wesbos/status/1681341649786486785"
                />
                <TestimonialCard
                  name="Justin Bennett"
                  quote="A super power of val.town is that the relative terseness of a val gives excellent API usage examples that are easy to grok. Want to learn atproto? There's a val for that. With a large enough corpus, it'll be the go to place for API examples."
                  avatar={justinbennet}
                  reference="Engineer @ Oxide, DevtoolsFM host, on Twitter"
                  // url="https://twitter.com/Zephraph/status/1672411315292323840?t=jtF1-DMU4Z4XiTyVQVTWxQ&s=19"
                />
                <TestimonialCard
                  name="pomdtr"
                  quote=" This is 🔥"
                  avatar={pomdtr}
                  reference="on Discord"
                  // url="https://discord.com/channels/1020432421243592714/1118643393455730721/1118645134658117753"
                />
                <TestimonialCard
                  name="0xferruccio"
                  quote="I'm mind-blown by http://val.town it's one of those products I can't believe didn't exist before"
                  avatar={_0xferruccio}
                  reference="on Twitter"
                  // url="https://discord.com/channels/1020432421243592714/1118643393455730721/1118645134658117753"
                />
              </div>
              <div className="hidden space-y-8 sm:block">
                <TestimonialCard
                  name="omgwtfbrblolttyl"
                  quote="yeah! working on replacing the backend of a personal project with a simple val related solution
        ive been waiting for something like valtown for building personal projects without getting bogged down in setting up toolchains and deploying and stuff"
                  avatar={omgwtf}
                  reference="on Discord"
                  // url="https://discord.com/channels/1020432421243592714/1123687468470915142/1123696228123811890"
                />
                <TestimonialCard
                  name="Ashik Ahmed"
                  quote="A practical middle option for not going too enterprisey (Lambda) or too consumer (zapier).
        "
                  avatar={ashikahmed}
                  reference="on Twitter"
                  // url="https://twitter.com/deputyashik/status/1669807993926598656"
                />
                <TestimonialCard
                  name="Nate Moore"
                  quote="@astrodotbuild 🤝 @ValDotTown...
            Really excited about Val's potential for persisted interaction on static sites!"
                  avatar={natemoore}
                  // url="https://twitter.com/n_moore/status/1682112004461404160?s=20"
                  reference="Astro co-creator on Twitter"
                />
                <TestimonialCard
                  name="Syntax"
                  quote="‘this is awesome’ ‘it's so cool’ ‘it's so neat’ - what @wesbos thinks about @ValDotTown from our episode with @stevekrouse"
                  avatar={syntax}
                  // url="https://twitter.com/n_moore/status/1682112004461404160?s=20"
                  reference="on Twitter"
                />
              </div>
              <div className="hidden space-y-8 sm:block">
                <TestimonialCard
                  name="Carl Von Blixen"
                  quote="What an absolute gem of a product! 💎 Amazing job @stevekrouse. I have wanted something like this (
@zapier for Javascript programmers - ish) for a LONG time."
                  reference="on Twitter"
                  avatar={carlvonblixen}
                  // url="https://twitter.com/carlvonblixen/status/1680104094927671296"
                />
                <TestimonialCard
                  name="infinitemap"
                  quote="I first setup a cloudflare worker email broker, but for some reason the worker was taking so long that my webhook was timing out, then I found val.town, it's such an amazing tool"
                  avatar={infinitemap}
                  // url="https://discord.com/channels/1020432421243592714/1121860024084467782/1121870889315737691"
                  reference="on Discord"
                />
                <TestimonialCard
                  name="Kyle Geske"
                  quote="val.town is bringing code alive, 
one () => {} at a time. 🔥"
                  avatar={kylegeske}
                  reference="on Twitter"
                  // url="https://twitter.com/stungeye/status/1671189310471835648?s=20"
                />
                <TestimonialCard
                  name="Jack Bridger"
                  quote="Val town is putting the fun back into coding. You can focus on the fun parts of just making magic happen without all the admin"
                  avatar={jackbridger}
                  reference="on Product Hunt"
                  // url="https://twitter.com/stungeye/status/1671189310471835648?s=20"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
```